apps/verticalhome/js/app.js

```diff
@@ -327,7 +403,35 @@
         // A hashchange event means that the home button was pressed.
         // The system app changes the hash of the homescreen iframe when it
         // receives a home button press.
+        case 'timeformatchange':
+          if (!this.l10nready) {
+            return;
+          }
+
+          this.timeFormat = window.navigator.mozHour12 ?
+            navigator.mozL10n.get('shortTimeFormat12') :
+            navigator.mozL10n.get('shortTimeFormat24');
+          this.refreshClock(new Date());
+
+          break;
         case 'hashchange':
+          this.backToMainScreen();
+          this.pinNavigation.reset();
+          this.clockTime.parentNode.style.opacity = 1;
+          this.clock.start(this.refreshClock.bind(this));
+
+
+          if (!this.l10nready) {
+            return;
+          }
+
+          this.timeFormat = window.navigator.mozHour12 ?
+            navigator.mozL10n.get('shortTimeFormat12') :
+            navigator.mozL10n.get('shortTimeFormat24');
+          this.refreshClock(new Date());
+
+
+
           // The group editor UI will be hidden by itself so returning...
           var editor = exports.groupEditor;
```

Comment:
I don't really know what the group editor UI is, but it sounds like code that could be removed.